var/spack/repos/builtin/packages/thepeg/package.py

```diff
@@ -71,7 +71,8 @@ class Thepeg(AutotoolsPackage):
     depends_on("hepmc3", when="hepmc=3")
     conflicts("hepmc=3", when="@:2.1", msg="HepMC3 support was added in 2.2.0")
     depends_on("fastjet", when="@2.0.0:")
-    depends_on("rivet", when="@2.0.3: +rivet")
+    depends_on("rivet@:3 hepmc=2", when="@2.0.3: +rivet hepmc=2")
```

Comment:
I think it would be useful to add maybe a fixme to indicate that rivet@4 support is still to be expected in a future thepeg version.

Reply:
Resolving since I added this in #48264 (which is an accidental duplicate for much of what's here already). `thepeg@2.3.0:` supports `rivet@4:`.